pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/SegmentPushUtils.java

```diff
@@ -88,7 +88,8 @@ public static URI generateSegmentTarURI(URI dirURI, URI fileURI, String prefix,
         return fileURI;
       }
     }
-    return URI.create(String.format("%s%s%s", prefix, fileURI.getRawPath(), suffix));
+    return URI.create(
+        String.format("%s%s%s", (prefix != null) ? prefix : "", fileURI.getRawPath(), (suffix != null) ? suffix : ""));
```

Comment:
Might be more readable
```suggestion
        (prefix != null ? prefix : "") + fileURI.getRawPath() + (suffix != null ? suffix : "");
```